fetch-todolist/src/component/Home.js

```diff
@@ -26,51 +29,61 @@ const Home = () => {
   };
 
   const handleDelete = (id) => {
-    const resltArray = list.filter((filteredItem) => id !== filteredItem.id);
-    setList(resltArray);
+    const resltArray = displayTodos.filter(
+      (filteredItem) => id !== filteredItem.id
+    );
+    setTodos(resltArray);
     openDelDiv();
   };
-  const openDiv = () => {
-    let condition = !isBlockHidden ? true : false;
-    setIsBlockHidden(condition);
-  };
   const openDelDiv = (id) => {
     let condition = !isDelBlockHidden ? true : false;
     setIsDelBlockHidden(condition);
     setIdToDel(id);
   };
-  const handleAdd = (task, ddate) => {
-    if (task !== "" && ddate !== "") {
-      const obj = {
-        id: list.length + 1,
-        task: task,
-        dueDtae: ddate.toString(),
-        status: false,
-      };
-      setList([...list, obj]);
-      setIsBlockHidden(false);
+  const handleStatusFilter = (status) => {
+    let completedTask = list;
+
+    if (status === "Completed") {
+      completedTask = completedTask.filter((task) => task.status);
+    } else if (status === "Active") {
+      completedTask = completedTask.filter((task) => !task.status);
     }
+    setTodos(completedTask);
+  };
+  const searchTask = (search) => {
+    let searchedList = list;
+    searchedList = searchedList.filter((item) => {
+      return item.task.includes(search);
+    });
+    setTodos(searchedList);
+  };
+  useEffect(() => {
```

Comment:
while writing the code maintain order of function type all use effect should be at end

Reply:
Done